app/models/application.rb

```diff
@@ -0,0 +1,35 @@
+class Application < ApplicationRecord
+  has_many :pet_applications, dependent: :destroy
+  has_many :pets, through: :pet_applications
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :description, presence: true
+  validates_inclusion_of :status, in: ["In Progress", "Pending", "Accepted", "Rejected"]
+
+  def get_pets
```

Comment:
No need to encapsulate this association in a model method. When we say `has_many :pets`, what Rails is doing under the hood is actually creating an instance method  called `pets` that will run a query to find all pets associated with this application. So we can call this `.pets` as if it's just another model method but it represents, specifically, our associations. 

You can just call `application.pets` in a controller or view